app_sensor_module/src/main.c

```diff
@@ -26,6 +27,41 @@ static const struct gpio_dt_spec led0 = GPIO_DT_SPEC_GET(LED0_NODE, gpios);
 static const struct gpio_dt_spec led1 = GPIO_DT_SPEC_GET(LED1_NODE, gpios);
 static const struct device *const wiznet = DEVICE_DT_GET_ONE(wiznet_w5500);
 
+static void ina_task(void *, void *, void *) {
+    struct sensor_value temperature;
+    power_module_telemetry_t sensor_telemetry = {0};
+    
+    const struct device *tmp116_device = DEVICE_DT_GET_ANY(ti_tmp116);
+    if (!device_is_ready(tmp116_device)) {
+        LOG_ERR("whomp whomp TMP116 sensor not found");
+        return;
+    }
+
+    // channels and storage for sensor readings
+    enum sensor_channel channels[] = {SENSOR_CHAN_AMBIENT_TEMP};
+    struct sensor_value *values[] = {&temperature};
+    l_sensor_readings_args_t args = {
+        .num_readings = 1,
+        .channels = channels,
+        .values = values, // Use if not converting to float
+        .float_values = {&some_float_var}, // float values
+    };
+    while (true) {
+        sensor_telemetry.timestamp = k_uptime_get_32();
+
+        // Fetch and read temperature data from TMP116 using aaron's helper function
+        int result = l_update_get_sensor_data(tmp116_device, &args, true); // set true if float conversion needed
+        if (result == 0) {
+            // If needed, convert sensor_value to float or double for processing
+            float temp_celsius = sensor_value_to_double(&temperature);
+            sensor_telemetry.data_temperature = temp_celsius;
+        }
```

Comment:
Shouldn't need to do this if you call a specific helper function to fetch the data as floats for you